Loop/View Models/SettingsViewModel.swift

```diff
@@ -81,7 +81,8 @@ public class SettingsViewModel: ObservableObject {
     let therapySettings: () -> TherapySettings
     let supportedInsulinModelSettings: SupportedInsulinModelSettings
     let pumpSupportedIncrements: (() -> PumpSupportedIncrements?)?
-    let syncPumpSchedule: (() -> PumpManager.SyncSchedule?)?
+    let syncBasalRateSchedule: (() -> PumpManager.SyncBasalRateSchedule?)?
+    let enactTempBasal: (() -> PumpManager.EnactTempBasal?)?
```

Comment:
I'm wondering if these method types (PumpManager.EnactTempBasal, and PumpManager.SyncBasalRateSchedule) are better reflected as specifications of the UI types that require them. For example, for changing maxTempBasal, the interaction between the settings UI and whoever is hosting it could be something like "user is changing max temp basal, are we good to go?", which would be better reflected with a function like `validateMaxTempChange(completion: (Error?) -> Void)`.  It would then be up to the host of the UI to decide what needs to happen on that call.

Reply:
@ps2 Yes, I was thinking of something similar, but not even more constrained; I was thinking maybe this method descriptor should instead be specifically scoped to the need here (e.g. something like `CancelTempBasal`), since that's what is needed.  Your solution is better, I think, because now it is clear that the callee _might_ cancel the temp basal, and might _not_. That said, I'm thinking that the signature should be:

```swift
typealias ValidateTempBasal = (_ unitsPerHour: Double, _ duration: TimeInterval, _ completion: @escaping (_ error: PumpManagerError?) -> Void) -> Void
```
(Basically the same signature) because I would imagine whatever code does the "validation" will need the proposed values (`unitsPerHour` and `duration`) to make that decision.  

I'll work on changing this.

I still have a question about the actual cancellation policy, and who should do it: 
1. Should it _only_ cancel if the unitsPerHour is _less than_ the current max basal? (or... if it is _less than the current _temp basal_?  That may be trickier to implement...)
2. In your opinion, what should have the responsibility for implementing this policy?  Right now I think it's in a bit of a weird place (the `DeliveryLimitsEditor`).

Reply:
Update from our discussion at dev sync:  Sounds like the answers to my questions are:
1. it should only cancel if the max basal is less than the current temp basal.
2. the policy should probably be in `LoopDataManager` or somewhere thereabouts